modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -358,6 +363,100 @@ public SCIMResponse listWithGET(UserManager userManager, String filter,
         }
     }
 
+    /**
+     * Build Node for filtering.
+     *
+     * @param filter Filter in the request.
+     * @param schema Schema
+     * @return Node
+     * @throws BadRequestException
+     * @throws IOException
+     */
+    private Node buildNode(String filter, SCIMResourceTypeSchema schema) throws BadRequestException, IOException {
+
+        if (filter != null) {
+            FilterTreeManager filterTreeManager = new FilterTreeManager(filter, schema);
+            return filterTreeManager.buildTree();
+        }
+        return null;
+    }
+
+    /**
+     * Resolves the sorting order of the filter.
+     *
+     * @param sortOrder Sort order in the request.
+     * @param sortBy    SortBy in the request.
+     * @return Resolved sorting order.
+     * @throws BadRequestException Invalid sorting order.
+     */
+    private String resolveSortOrder(String sortOrder, String sortBy) throws BadRequestException {
+
+        if (sortOrder != null) {
+            if (!(sortOrder.equalsIgnoreCase(SCIMConstants.OperationalConstants.ASCENDING) || sortOrder
+                    .equalsIgnoreCase(SCIMConstants.OperationalConstants.DESCENDING))) {
+                String error = " Invalid sortOrder value is specified";
+                throw new BadRequestException(error, ResponseCodeConstants.INVALID_VALUE);
+            }
+        }
+        // If a value for "sortBy" is provided and no "sortOrder" is specified, "sortOrder" SHALL default to
+        // ascending.
+        if (sortOrder == null && sortBy != null) {
+            return SCIMConstants.OperationalConstants.ASCENDING;
+        }
+        return sortOrder;
+    }
+
+    /**
+     * Method to process a user list and return a SCIM response.
+     *
+     * @param tempList          Filtered user list
+     * @param encoder           Json encoder
+     * @param schema            Schema
+     * @param attributes        Required attributes
+     * @param excludeAttributes Exclude attributes
+     * @param startIndex        Starting index
+     * @return SCIM response
+     * @throws NotFoundException
+     * @throws CharonException
+     * @throws BadRequestException
+     */
+    private SCIMResponse processUserList(List<Object> tempList, JSONEncoder encoder, SCIMResourceTypeSchema schema,
+            String attributes, String excludeAttributes, int startIndex)
+            throws NotFoundException, CharonException, BadRequestException {
+
+        int totalResults = 0;
+        List<Object> returnedUsers;
+        if (tempList == null) {
+            tempList = Collections.emptyList();
+        }
+        try {
+            totalResults = (int) tempList.get(0);
+            tempList.remove(0);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
instead of catching  IndexOutOfBoundsException, shall we use CollectionUtils.isEmpty at the beginning?

Reply:
apache-commons is not there in Charon. Therefore, unable to use CollectionUtils.